scipy/optimize/_minimize.py

```diff
@@ -597,53 +597,89 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
                       bounds=bounds, constraints=constraints,
                       callback=callback, **options)
 
+    if constraints is not None:
+        constraints = standardize_constraints(constraints, x0, meth)
+
+    remove_vars = False
     if bounds is not None:
         bounds = standardize_bounds(bounds, x0, meth)
 
-    if constraints is not None:
-        constraints = standardize_constraints(constraints, x0, meth)
+        lb, ub = _split_bounds(bounds)
+        btol = 1e-8  # tolerance for bounds being considered equal. Expose?
+        i_fixed = np.abs(ub - lb) < btol
+        remove_vars = i_fixed.any() and meth in {'powell', 'l-bfgs-b', 'slsqp'}
+
+    if remove_vars:
+        x_fixed = ((lb+ub)/2)[i_fixed]  # values of fixed variables
+        x0 = x0[~i_fixed]   # remove fixed variables from x0
+        bounds = _remove_from_bounds(bounds, i_fixed)
+        fun = _remove_from_func(fun, i_fixed, x_fixed)
+        if callable(jac):
+            jac = _remove_from_func(jac, i_fixed, x_fixed, remove=1)
+        if callable(hess):
+            hess = _remove_from_func(hess, i_fixed, x_fixed, remove=2)
+        if callable(callback):
+            callback = _remove_from_func(callback, i_fixed, x_fixed)
+        for con in constraints:
+            if isinstance(con, dict):
```

Comment:
what about if the original con is a `NonLinearConstraint`, or is it ok because the constraints are normalised above.

Reply:
Is constraints always a sequence, it might be a single constraint.

Reply:
> what about if the original con is a NonLinearConstraint

Should be OK because things get mapped to the old style by `standardize_constraints`

> Is constraints always a sequence, it might be a single constraint.

True, should consider that.

Reply:
> Should be OK because things get mapped to the old style by standardize_constraints

That's ok for SLSQP, not for `trust-constr`.

SLSQP uses [`approx_derivative` for constraint jacobians](https://github.com/scipy/scipy/blob/master/scipy/optimize/slsqp.py#L295), unless the constraint has a callable jacobian. It therefore respects bounds provided to it.

TNC uses `approx_derivative` as [well](https://github.com/scipy/scipy/blob/master/scipy/optimize/tnc.py#L375) because a `ScalarFunction` is created in `_prepare_scalar_function`. I think all minimize methods use `approx_derivative` now.

`newton_cg` requires a callable Jacobian.

Reply:
>  I think all minimize methods use `approx_derivative` now

Well, except `newton_cg`, apparently. And it sounds like if they're using it, there's no way to pick the approximation used; it just defaults to something. OK.